### PR TITLE
Improve ESM support

### PR DESCRIPTION
**Pull request by lpellegr**
Following the PR for [orderedmap](https://github.com/marijnh/orderedmap/pull/5), this one shows the changes I would like to introduce to support PM in buildless environments.

I took the opportunity to upgrade Rollup dependencies and improve the packaging by keeping only the required files.

PS: if you release _orderedmap_ as a new major version (although I think a minor version should be enough since no breaking changes are introduced) then the version defined in this package must be bumped.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 /node_modules
 .tern-port
 /dist
+
+*.iml
```
Comment (marijnh) [inline .gitignore]: Please don't mix random tweaks related to your build environment into pull requests. See also the various things added to `.npmignore`.

**Comment by marijnh** [comment]
If you can open a PR to all submodules with the rollup upgrade and module field, that'd be useful. Please don't remove the `./` in front of the input file in rollup.config.js—those are there so that the `external` filter doesn't fire for the main input, and removing it breaks the build.

Also please leave out 80d842f , I don't want to try to create ignore files that work with all possible tools.

**Comment by lpellegr** [comment]
Thanks for the review. 

I have updated the PR with your suggestions.

I will continue with the submodules and the few transitive dependencies that are required.

**Comment by marijnh** [comment]
Looks good now, except for the semicolons in the rollup config file (the project uses semicolon-less style).

If you could submit a PR like this for all the submodules, I'll merge them. The external modules should now all be module-enabled, so you can bump those versions.

`markdown-it`, which is a dependency of the prosemirror-markdown module, isn't though. I'd still like to upgrade rollup for that one, even if it can't be loaded as a module yet, just so that a mutli-repo install only requires one rollup version.

**Comment by lpellegr** [comment]
Ok, thanks. Changes have been applied to semicolons. PRs for submodules are coming.

**Comment by pocke** [comment]
Hi. Thank you for the work for ESM.
But it does not work for our application, so we stop upgrading prosemirror-* packages.

We use ES5, but the published ESM file has ES6 features such as `const`. So our build process is failing with the latest prosemirror packages.


Could you add a build script for ESM and specify the built ESM file as a `"module"` in package.json instead of `src/index.js`?
I think it is the mainstream way, because I can only find ES5 code from files that are specified as `module` in our application.

**Comment by pocke** [comment]
related pull request: https://github.com/ProseMirror/prosemirror-markdown/pull/36

**Pull request merged**